electrumx/server/session.py

```diff
@@ -213,29 +216,158 @@ def _ssl_context(self):
     async def _start_servers(self, services):
         for service in services:
             kind = service.protocol.upper()
-            if service.protocol in self.env.SSL_PROTOCOLS:
-                sslc = self._ssl_context()
-            else:
-                sslc = None
-            if service.protocol == 'rpc':
-                session_class = LocalRPC
-            else:
-                session_class = self.env.coin.SESSIONCLS
-            if service.protocol in ('ws', 'wss'):
-                serve = serve_ws
-            else:
-                serve = serve_rs
-            # FIXME: pass the service not the kind
-            session_factory = partial(session_class, self, self.db, self.mempool,
-                                      self.peer_mgr, kind)
-            host = None if service.host == 'all_interfaces' else str(service.host)
-            try:
-                self.servers[service] = await serve(session_factory, host,
-                                                    service.port, ssl=sslc)
-            except OSError as e:    # don't suppress CancelledError
-                self.logger.error(f'{kind} server failed to listen on {service.address}: {e}')
+            if service.protocol == 'http':
+                host = None if service.host == 'all_interfaces' else str(service.host)
+                try:
+                    app = web.Application(middlewares=[
+                        error_middleware(self),
+                        request_middleware(self)
+                    ])
+                    handler = HttpHandler(self, self.db, self.mempool, self.peer_mgr, kind)
+                    # GET
+                    app.router.add_get('/proxy', handler.proxy)
+                    app.router.add_get('/proxy/health', handler.health)
+                    app.router.add_get('/proxy/blockchain.block.header', handler.block_header)
```

Comment:
Do we need `/proxy` at the front?  

The service is started at say some ip address, then the HTTP would be served at the root, which would be fine, right?

Reply:
I retained the /proxy prefix to maintain complete compatibility with the input and output formats used when originally using electrumx-proxy. We have deployed this code in a production environment and achieved seamless transition. Even when the entire network nodes were slow and unusable yesterday, our service continued to run stably.